### PR TITLE
Cleanups: np.clip and np.ptp are awesome

### PR DESCRIPTION
**Pull request by anntzer**
All's in the title.

Also clarified that `draw_rubberband` does not require its arguments to be ordered... tested on all backends except on gtk, which seems broken as of master for me, and macos, which I cannot test and for which I would appreciate a confirmation (see https://gitter.im/matplotlib/matplotlib?at=5831157a238757566ccb3580).

### CODE REVIEW

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -3013,20 +3016,13 @@ def drag_zoom(self, event):
         if self._xypress:
             x, y = event.x, event.y
             lastx, lasty, a, ind, view = self._xypress[0]
-
-            # adjust x, last, y, last
-            x1, y1, x2, y2 = a.bbox.extents
-            x, lastx = max(min(x, lastx), x1), min(max(x, lastx), x2)
-            y, lasty = max(min(y, lasty), y1), min(max(y, lasty), y2)
-
+            (x1, y1), (x2, y2) = np.clip(
```
Comment (tacaswell) [inline lib/matplotlib/backend_bases.py]: This is on a hot-path (mouse move event callback), might be worth micro-benchmarking this.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -3675,8 +3675,7 @@ def dopatch(xs, ys, **kwargs):
 
         # width
         if widths is None:
-            distance = max(positions) - min(positions)
-            widths = [min(0.15 * max(distance, 1.0), 0.5)] * N
+            widths = [np.clip(np.ptp(positions), 0.15, 0.5)] * N
```
Comment (anntzer) [inline lib/matplotlib/axes/_axes.py]: Good catch.  Fixed, and updated the docstring to reflect the actual behavior (which was misdocumented).

**Comment by QuLogic** [comment]
This needs some small conflict resolution.

**Comment by anntzer** [comment]
Fixed.
Thanks for looking at this.

**Comment by anntzer** [comment]
smuggled in an additional cleanup (duplicated line, see last commit)

**Comment by QuLogic** [comment]
Conflicts (from your other PR?)

**Comment by anntzer** [comment]
rebased

**Pull request merged**